### PR TITLE
Feature/support advanced ao filters

### PR DESCRIPTION
**Pull request by anthonygarvan**

### CODE REVIEW

File: openfecwebapp/api_caller.py [hunk 1]
```diff
@@ -49,14 +49,21 @@ def load_search_results(query, query_type='candidates'):
...
+def load_legal_search_results(query, query_type='all', ao_no=None, ao_name=None,
+                                offset=0, limit=20):
     filters = {}
     if query:
         filters['q'] = query
         filters['hits_returned'] = limit
         filters['type'] = query_type
         filters['from_hit'] = offset
 
+        if ao_no and ao_no[0]:
```
Comment (vrajmohan) [inline openfecwebapp/api_caller.py]: This idiom is unclear to me. Why does `if ao_no` not suffice? What are we guarding against?

Comment (anthonygarvan) [inline openfecwebapp/api_caller.py]: _sighs_. So, the standard HTML form sends back empty keywords, but we are interpreting them as a list, so we get back results that look like `['']`: a list with a single empty string as an element.

**Comment by anthonygarvan** [comment]
@vrajmohan I think this is good to go now, pls merge if it's ok with you.

File: static/js/pages/legal.js [hunk 2]
```diff
@@ -1,10 +1,15 @@
 'use strict';
 
 /* global require */
+require('jquery.inputmask');
+require('jquery.inputmask/dist/inputmask/inputmask.date.extensions.js');
+require('jquery.inputmask/dist/inputmask/inputmask.numeric.extensions.js');
```
Comment (noahmanger) [inline static/js/pages/legal.js]: Responding to @vrajmohan 's question on my previous PR,`inputmask.numeric.` isn't required. Copy-and-paste mistake. :)

Comment (anthonygarvan) [inline static/js/pages/legal.js]: thanks! removed.

**Pull request merged**